milli/src/search/new/ranking_rule_graph/build.rs

```diff
@@ -0,0 +1,86 @@
+use super::{Edge, RankingRuleGraph, RankingRuleGraphTrait};
+use crate::search::new::interner::{DedupInterner, MappedInterner};
+use crate::search::new::query_graph::{QueryNode, QueryNodeData};
+use crate::search::new::small_bitmap::SmallBitmap;
+use crate::search::new::{QueryGraph, SearchContext};
+use crate::Result;
+use std::collections::HashSet;
+
+impl<G: RankingRuleGraphTrait> RankingRuleGraph<G> {
+    /// Build the ranking rule graph from the given query graph
+    pub fn build(
+        ctx: &mut SearchContext,
+        query_graph: QueryGraph,
+        cost_of_ignoring_node: MappedInterner<QueryNode, Option<(u32, SmallBitmap<QueryNode>)>>,
+    ) -> Result<Self> {
+        let QueryGraph { nodes: graph_nodes, .. } = &query_graph;
```

Comment:
TODO: documentation: while this function is available for every `RankingRuleGraph<G>` and relies on the `build_edges` method, we should say somewhere that it is completely fine to:
1. replace the `build_edges` by a `build_graph` method
2. provide this function as a convenience function for ranking rule graphs that have a `build_edges` function